src/core/integrate.cpp

```diff
@@ -241,37 +260,29 @@ void integrate_vv(int n_steps, int reuse_forces) {
 
 #endif
 
-    /* Integration Steps: Step 1 and 2 of Velocity Verlet scheme:
-       v(t+0.5*dt) = v(t) + 0.5*dt * a(t)
-       p(t + dt)   = p(t) + dt * v(t+0.5*dt)
-       NOTE: Depending on the integration method Step 1 and Step 2
-       cannot be combined for the translation.
-    */
-    if (integ_switch == INTEG_METHOD_NPT_ISO) {
-      propagate_vel(particles);
-      propagate_pos(particles);
-
-      /* Propagate time: t = t+dt */
-      sim_time += time_step;
-    } else if (integ_switch == INTEG_METHOD_STEEPEST_DESCENT) {
+    if (integ_switch == INTEG_METHOD_STEEPEST_DESCENT) {
```

Comment:
I think this could be made clearer here by introducing an other level of abstraction:
What we are doing here is running a two-step integration scheme, the actual implementation
is determined by `integ_switch`. Subsequently there should be a function here that is called
`step_1` or `intgrator_step1` or similar, that den switches on `integ_switch`. Then, after the
force calc there should be the corresponding `step_2` function. The fact that for VV the steps
are divided further is of no concern to the called. For one-step integrators, like steepest descent,
the 2nd step is a noop.